test/common/http/conn_manager_impl_test.cc

```diff
@@ -1069,64 +1069,6 @@ TEST_F(HttpConnectionManagerImplTest, DoNotStartSpanIfTracingIsNotEnabled) {
   conn_manager_->onData(fake_input, false);
 }
 
-TEST_F(HttpConnectionManagerImplTest, StartSpanOnlyHealthCheckRequest) {
```

Comment:
Why was this test removed?

Reply:
The current implementation avoids sampling the initial span by not finishing it.

However if a child span is created from that initial span (as in the case of the router setting `start_child_span` to true, then the child span is reported with a parent span id that does not exist (on the server anyway).

So this PR uses the sampling mechanism to disable reporting of the span, which is then also propagated to the child spans.